### PR TITLE
Refs #4478 - API doc strings marked for translation

### PR DESCRIPTION
**Pull request by mbacovsky**

### CODE REVIEW

**Comment by domcleal** [comment]
@mbacovsky there are a couple of test failures due to the changed messages, but this one's puzzling me, I can't think why it'd change: http://ci.theforeman.org/job/test_develop_pr_core/1723/database=mysql,puppet=3.0,ruby=1.9.3/testReport/junit/%28root%29/Api__V2__HostsControllerTest/test_0029_when_a_bad__type_is_requested___unprocessable_entity_is_returned/

**Comment by mbacovsky** [comment]
Same here. The method is returning: `ERF42-3624 [Foreman::Exception]: A problem occurred when detecting host type: uninitialized constant Host::Invalid` instead of expected `ERF42-2640`. Investigating further...

File: app/controllers/api/v2/hosts_controller.rb [hunk 1]
```diff
@@ -199,10 +197,10 @@ def detect_host_type
           logger.debug "Creating host of type: #{params[:type]}"
           return params[:type].constantize
         else
-          raise "Invalid type for host creation via facts: #{params[:type]}"
+          raise ::Foreman::Exception.new(N_("Invalid type for host creation via facts: %s"), params[:type])
         end
       rescue => e
-        raise ::Foreman::Exception.new("A problem occurred when detecting host type: #{e.message}")
+        raise ::Foreman::Exception.new(N_("A problem occurred when detecting host type: %s"), e.message)
```
Comment (mbacovsky) [inline app/controllers/api/v2/hosts_controller.rb:203]: @domcleal , the formatting of the message changed. The `e.message` is now parameter of the message and not its part. Error codes are computed from the message without parameters so the code for this message is different. I fixed it in the tests. However I'm not sure if this is proper fix or if it violates some error codes rules somehow.

Comment (domcleal) [inline app/controllers/api/v2/hosts_controller.rb:203]: @mbacovsky ah ok, thanks.  I think that's unavoidable for now, let's just change it.  Maybe one day we can implement some kind of override for error codes so we can keep them static.

**Comment by domcleal** [comment]
Thanks @mbacovsky, merged as 2656873 and Transifex updated.

File: app/controllers/api/v2/hosts_controller.rb (hunk 1 above)
Comment (lzap) [inline app/controllers/api/v2/hosts_controller.rb:203]: This case is a simple one - it was wrong and Martin fixed it, therefore the code must change. I'd rather fix the error code on the wiki than introducing overridable codes.

**Pull request closed without merging**